### PR TITLE
fix(design-system): converge artists page text-size arbitraries to tokens (−13)

### PR DESCRIPTION
**Pull request by itstimwhite**
## Summary

Converges `app/artists/page.tsx` from arbitrary Tailwind text-size values to canonical design-system token utilities. One file, 13 replacements, no visual change.

**Replacements made:**

| Before | After | Count | px |
|--------|-------|-------|-----|
| `text-[12px]` | `text-xs` | ×2 | 12px |
| `text-[13px]` | `text-app` | ×9 | 13px |
| `text-[15px]` | `text-mid` | ×2 | 15px |

**Left unchanged (no exact token equivalent):**
- `tracking-[-0.03em]` — custom negative tracking on stat count headline
- `tracking-[0.14em]` — positive tracking on uppercase label (no canonical token)

## Ratchet

| Metric | Value |
|--------|-------|
| Baseline before (main after multiple convergence PRs) | 5394 |
| Actual count after this PR (rebased on current main) | 5066 |
| Baseline after | 5066 |
| This PR's direct removals | −13 |

Note: The baseline drops from 5394 → 5066 because several PRs merged to main since the previous rebase reduced the actual count without updating the baseline. This PR sets it to the tightest correct value (verified by grep count on the rebased working tree).

File: `apps/web/tests/unit/design-system/arbitrary-values.baseline.json` updated in same commit.

## Verification

- [x] TypeScript: `pnpm --filter @jovie/web run typecheck` — 0 errors
- [x] Ratchet: actual count 5066 ≤ baseline 5066 — PASS
- [x] Token validity: `text-xs` (0.75rem/12px), `text-app` (0.8125rem/13px), `text-mid` (0.9375rem/15px) confirmed in `design-system.css` `@theme`
- [x] Rebased onto current main (`6706bbe`); baseline conflict resolved to exact grep count

🤖 Scheduled design-system convergence run · 2026-06-20

### CODE REVIEW

**Comment by itstimwhite** [comment]
### Merge activity

* **Jun 20, 3:19 PM UTC**: The merge label 'fast' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=JovieInc&repo=Jovie) once it meets the requirements.
* **Jun 20, 4:16 PM UTC**: `itstimwhite` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=JovieInc&repo=Jovie).
* **Jun 20, 5:55 PM UTC**: CI is running for this pull request on a draft pull request ([#11279](https://app.graphite.com/github/pr/JovieInc/Jovie/11279)) due to your merge queue CI optimization settings.
* **Jun 20, 6:07 PM UTC**: CI is running for this pull request on a draft pull request ([#11282](https://app.graphite.com/github/pr/JovieInc/Jovie/11282)) due to your merge queue CI optimization settings.
* **Jun 20, 6:31 PM UTC**: The [Graphite merge queue](https://app.graphite.com/merges?org=JovieInc&repo=Jovie) couldn't merge this PR because **it had merge conflicts**.
* **Jun 20, 6:33 PM UTC**: The merge label 'fast' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=JovieInc&repo=Jovie) once it meets the requirements.
* **Jun 20, 7:05 PM UTC**: The merge label 'fast' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=JovieInc&repo=Jovie) once it meets the requirements.

**Comment by itstimwhite** [comment]
⚠️ Merge Conflict Detected — This PR has state 'BLOCKED' and cannot be auto-merged.

Branch: 'tim/jov-ds-drift-artists-page'
State: BLOCKED



To resolve:
1. git fetch origin main
2. git checkout tim/jov-ds-drift-artists-page
3. git rebase origin/main
4. Fix any conflicts, then git rebase --continue
5. git push --force-with-lease origin tim/jov-ds-drift-artists-page

This comment will not repeat for 2 hours.

**Comment by itstimwhite** [comment]
## Review: PR #11263

### PASS 1 — Logic, Security, Bugs

**Logic errors:** None found. Purely a stylistic token swap — no control flow, conditionals, or data transformations changed.

**Security:** None found. No user input handling, no auth changes, no DB queries, no external HTTP calls, no CSP modifications.

**Data integrity:** None found. No DB queries, no mutations, no schema changes, no migrations.

**Error handling:** None found. No new code paths, no async operations, no error surfaces introduced.

**Cross-file impact:** None found. The only non-style file changed is the baseline JSON fixture, which is updated to match the new count. The token classes (`text-xs`, `text-app`, `text-mid`) are confirmed to already exist in the design system and are used across 1300+ components, so no breakage risk.

---

### PASS 2 — Design System & Code Quality

**Token correctness:** All three replacements map 1:1 to existing `@theme` tokens:
- `text-[12px]` → `text-xs` (0.75rem = 12px) ✓
- `text-[13px]` → `text-app` (0.8125rem = 13px) ✓
- `text-[15px]` → `text-mid` (0.9375rem = 15px) ✓

**Baseline update:** The baseline count drops from 6375 to 5406 (−13), matching the 13 replacements. The PR description notes the old baseline was stale (true live count was 5419), so this both corrects the stale baseline and applies the −13 reduction. This is consistent.

**Custom values left intact:** `tracking-[-0.03em]` and `tracking-[0.14em]` are correctly left as-is since no canonical tracking tokens exist for those values.

**No visual change claimed and verified:** The PR states no visual change, and the 1:1 px-equivalent mappings support this.

**No emoji, no native dialogs, no decorative hover motion:** Not applicable — no such patterns introduced.

---

### PASS 3 — Requirements Coverage

The PR description is thorough: it enumerates every replacement, explains what was left unchanged and why, provides ratchet metrics, and lists verification steps (typecheck, ratchet test, token confirmation). All checkboxes are marked complete.

---

### Verdict

**Approve.** Mechanically correct, zero-risk token convergence. No issues found across any pass.

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot v2 (openrouter/owl-alpha).
This review uses GBrain knowledge base context + multi-pass analysis.
React with 👍 if helpful, 👎 if not — helps me improve.

**Comment by itstimwhite** [comment]
## Review: PR #11263

### PASS 1 — Logic, Security, Bugs

**Logic errors:** None found. Purely mechanical replacement of arbitrary Tailwind values with equivalent token utilities. No conditional logic, data flow, or control flow changes.

**Security:** None found. No user input handling, no auth changes, no data fetching, no HTML injection surface. The `profile.bio` and `profile.displayName` interpolations were already present before this diff — this PR doesn't touch them.

**Data integrity:** None found. No DB queries, no mutations, no schema changes.

**Error handling:** None found. No async operations or error paths introduced or modified.

**Cross-file impact:** None found. The only non-page file changed is the baseline JSON fixture, which is updated to match the new count. The token utilities (`text-xs`, `text-app`, `text-mid`) are already defined in `design-system.css` and used across 1300+ components — no new dependencies introduced.

---

### PASS 2 — Design System & Code Quality

**Token correctness:** All three replacements are verified correct:

| Token | `@theme` value | Matches |
|-------|---------------|---------|
| `text-xs` | 0.75rem (12px) | `text-[12px]` ✓ |
| `text-app` | 0.8125rem (13px) | `text-[13px]` ✓ |
| `text-mid` | 0.9375rem (15px) | `text-[15px]` ✓ |

**Custom values correctly retained:** `tracking-[-0.03em]` and `tracking-[0.14em]` have no canonical token equivalent — leaving them as arbitrary values is the correct call per the PR description.

**Baseline update:** The ratchet baseline drops from 6375 → 5406 (−13), matching the 13 replacements exactly. The PR description notes the old baseline was stale (true live count was 5419); the new baseline of 5406 is consistent with 5419 − 13 = 5406. ✓

**Verification checks pass:** TypeScript clean, ratchet test passes, token definitions confirmed.

---

### PASS 3 — Requirements Coverage

The PR does exactly what it claims: converges 13 arbitrary text-size values in `app/artists/page.tsx` to canonical design-system tokens, updates the ratchet baseline, and makes no visual or behavioral change. No scope creep, no unrelated changes.

---

### Summary

**No issues found.** This is a clean, well-scoped design-system convergence PR. All replacements are 1:1 equivalent, the baseline math checks out, and the custom tracking values are correctly left as-is. Approve.

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot v2 (openrouter/owl-alpha).
This review uses GBrain knowledge base context + multi-pass analysis.
React with 👍 if helpful, 👎 if not — helps me improve.

**Comment by itstimwhite** [comment]
## Review: PR #11263

### PASS 1 — Logic, Security, Bugs

**Logic errors:** None found. This is a pure styling substitution — replacing 13 arbitrary Tailwind text-size values with equivalent design-token utilities. No logic paths, conditionals, or data flow are touched.

**Security:** None found. No user input handling, no auth changes, no DB queries, no external HTTP calls, no CSP modifications. Purely presentational class swaps.

**Data integrity:** None found. No DB queries, no mutations, no schema changes, no migrations.

**Error handling:** None found. No async operations, no try/catch paths, no promise chains introduced or modified.

**Cross-file impact:** None found. The only non-`page.tsx` change is the baseline JSON fixture, which is updated to match the new count. The token utilities (`text-xs`, `text-app`, `text-mid`) are already established in the design system and used across 1300+ components, so no new dependency or compatibility risk is introduced.

---

### PASS 2 — Design System & Code Quality

**Token correctness:** All three replacements are verified correct:

| Token | `@theme` value | Matches |
|-------|---------------|---------|
| `text-xs` | 0.75rem / 12px | `text-[12px]` ✓ |
| `text-app` | 0.8125rem / 13px | `text-[13px]` ✓ |
| `text-mid` | 0.9375rem / 15px | `text-[15px]` ✓ |

**Custom values correctly retained:** `tracking-[-0.03em]` and `tracking-[0.14em]` have no canonical token equivalent and are appropriately left as arbitrary values. This is correct — the PR description explicitly calls this out.

**Baseline update:** The ratchet baseline drops from 6375 → 5406 (−13), matching the 13 replacements. The PR description notes the old baseline was stale (true live count was 5419), so this both corrects the stale baseline and applies the −13 delta. The ratchet test passes at count ≤ baseline.

**No visual change:** Confirmed — every replacement is a 1:1 px-equivalent swap.

**Code quality:** Clean, focused, single-file change. No `any`, no type issues, no formatting concerns.

---

### PASS 3 — Requirements & Completeness

**Scope:** The PR does exactly what it claims — converges 13 arbitrary text-size values in `app/artists/page.tsx` to design tokens. No scope creep, no unrelated changes.

**Baseline fixture updated:** Yes, in the same commit. Ratchet test passes.

**Verification checklist:** All three items (typecheck, ratchet, token validity) are confirmed in the PR description.

**Edge cases:** The two custom `tracking-*` values are correctly left alone. The `&apos;` entity in "artist's" is pre-existing and untouched. The `group-hover:text-secondary-token` on the `h2` at the artist card headline is a pre-existing pattern, not introduced here.

---

### Summary

**No issues found.** This is a clean, correct, well-scoped design-system convergence PR. All 13 substitutions are 1:1 px-equivalent token swaps, the baseline is updated consistently, and the two custom tracking values are correctly retained. Approved.

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot v2 (openrouter/owl-alpha).
This review uses GBrain knowledge base context + multi-pass analysis.
React with 👍 if helpful, 👎 if not — helps me improve.

**Comment by itstimwhite** [comment]
## Review: PR #11263

### PASS 1 — Logic, Security, Bugs

**Logic errors:** None found. Purely mechanical find-and-replace of Tailwind class strings. No control flow, conditionals, or data transformations touched.

**Security:** None found. No user input handling, no auth changes, no DB queries, no HTTP calls, no CSP modifications.

**Data integrity:** None found. No DB queries, no mutations, no schema changes, no migrations.

**Error handling:** None found. No async logic, no try/catch paths, no promise chains introduced or modified.

**Cross-file impact:** None found. The only non-page file changed is the baseline JSON fixture, which is updated to match the new count. No imports, exports, shared types, or component contracts were altered.

---

### PASS 2 — Design System & Code Quality

**Token validity:** All three replacements are confirmed canonical tokens:
- `text-xs` → 0.75rem / 12px ✓
- `text-app` → 0.8125rem / 13px ✓
- `text-mid` → 0.9375rem / 15px ✓

**Custom values correctly retained:**
- `tracking-[-0.03em]` on the stat count headline — no canonical tracking token exists for this value. Correctly left as-is.
- `tracking-[0.14em]` on the uppercase "Public profiles" label — same reasoning. Correct.

**Baseline update:** The ratchet baseline drops from 6375 → 5406 (−13), matching the 13 replacements exactly. The PR description acknowledges the prior baseline was stale (true live count was 5419), so this is a correction of the baseline *plus* the 13 new removals. The ratchet test passes at count ≤ 5406.

**No visual change:** All replacements are 1:1 px-equivalent swaps. No layout, spacing, or interaction changes.

**Commit hygiene:** Single-file convergence with baseline update in the same commit. Conventional commit format `fix(design-system): ...` is appropriate.

---

### PASS 3 — Requirements & Completeness

**Scope:** The PR claims 13 replacements in one file. Counting the diff: 2× `text-[12px]` → `text-xs`, 9× `text-[13px]` → `text-app`, 2× `text-[15px]` → `text-mid`. That's 13. ✓

**Verification checklist:** TypeScript typecheck (0 errors), ratchet test (PASS), token confirmation in `@theme` — all checked. ✓

**No partial work:** Every `text-[12px]`, `text-[13px]`, and `text-[15px]` instance in this file appears to have been replaced. No stragglers visible in the diff.

---

### Summary

Clean, mechanical design-system convergence. No logic, security, or quality issues. Baseline ratchet is consistent with the change count. Ready to merge.

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot v2 (openrouter/owl-alpha).
This review uses GBrain knowledge base context + multi-pass analysis.
React with 👍 if helpful, 👎 if not — helps me improve.

File: apps/web/tests/unit/design-system/arbitrary-values.baseline.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "count": 6375,
+  "count": 5392,
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/tests/unit/design-system/arbitrary-values.baseline.json]: This suggestion is incorrect — `5392` is the right value. The bot's reasoning doesn't account for PR #11244 (EntityCard convergence) which merged to `main` before this branch was rebased.

Correct timeline:
- Live count before #11244: ~5419
- #11244 merged → removed 14 arbitrary values → live count: **5405**
- This PR removes 13 more → live count: **5392**
- Ratchet assertion after merge: `5392 ≤ 5392` → ✅

The `5406` figure the bot expects is `5419 − 13`, which would have been correct had #11244 not landed first. Setting the baseline to `5406` would incorrectly allow 14 regressions to creep back in. The `5392` value was verified by running the actual grep count on the rebased working tree.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by itstimwhite** [comment]
## PASS 1 — Logic, Security, Bugs

### Logic errors
None found. This is a pure CSS class substitution — replacing arbitrary Tailwind values (`text-[12px]`, `text-[13px]`, `text-[15px]`) with equivalent token utilities (`text-xs`, `text-app`, `text-mid`). No conditional logic, loops, or data flow changes.

### Security
None found. No user input handling, no auth changes, no DB queries, no HTTP calls, no CSP modifications. Purely presentational class swaps.

### Data integrity
None found. No DB queries, no mutations, no schema changes, no validation logic touched.

### Error handling
None found. No async operations, no try/catch blocks, no promise chains introduced or modified.

### Cross-file impact
None found. The only non-`page.tsx` change is the baseline JSON count update, which is a test fixture — not a runtime dependency. No imports, exports, or shared types were modified. The token utilities (`text-xs`, `text-app`, `text-mid`) are already defined in `design-system.css` and used across 1300+ components, so no new CSS needs to be generated.

---

**Summary:** Clean pass. Zero logic, security, or data-integrity concerns. This is a straightforward design-token convergence with no behavioral surface area.

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot v2 (openrouter/owl-alpha).
This review uses GBrain knowledge base context + multi-pass analysis.
React with 👍 if helpful, 👎 if not — helps me improve.

**Comment by itstimwhite** [comment]
⚠️ Merge Conflict Detected — This PR has state 'DIRTY' and cannot be auto-merged.

Branch: 'tim/jov-ds-drift-artists-page'
State: DIRTY



To resolve:
1. git fetch origin main
2. git checkout tim/jov-ds-drift-artists-page
3. git rebase origin/main
4. Fix any conflicts, then git rebase --continue
5. git push --force-with-lease origin tim/jov-ds-drift-artists-page

This comment will not repeat for 2 hours.

**Comment by itstimwhite** [comment]
## PASS 1 — Logic, Security, Bugs

### Logic errors
None found. This is a pure CSS class substitution — replacing arbitrary Tailwind values (`text-[12px]`, `text-[13px]`, `text-[15px]`) with equivalent token utilities (`text-xs`, `text-app`, `text-mid`). No control flow, conditionals, or data transformations are touched.

### Security
None found. No user input handling, no auth logic, no DB queries, no external HTTP calls, no CSP changes. Purely presentational class swaps.

### Data integrity
None found. No DB queries, no mutations, no schema changes, no validation logic altered.

### Error handling
None found. No async logic, no try/catch paths, no promise chains introduced or modified.

### Cross-file impact
None found. The only non-`page.tsx` change is the baseline JSON ratchet (count 6361 → 5392), which is a test fixture that exists solely to track this convergence effort. No other files import from or depend on the class names in `artists/page.tsx`.

---

**Summary:** Clean pass on Pass 1. This is a straightforward design-system token convergence — 13 class replacements in one component file plus a baseline count decrement. No logic, security, data, or cross-file concerns.

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot v2 (openrouter/owl-alpha).
This review uses GBrain knowledge base context + multi-pass analysis.
React with 👍 if helpful, 👎 if not — helps me improve.

File: apps/web/tests/unit/design-system/arbitrary-values.baseline.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
-  "count": 5394,
+  "count": 5066,
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/tests/unit/design-system/arbitrary-values.baseline.json:2]: Not a real bug — same pattern as the earlier false positive on this PR (see above thread, now outdated).

The bot's `5392` figure is from the **previous rebase iteration** (branch on main after PR #11244 only). Since then, five more PRs merged to main (`#11099`, `#11108`, `#11160`, `#11202`, `#11234`), each independently reducing the arbitrary value count. The current branch HEAD (`41c29b8`) is rebased on main at `6706bbe`, which reflects all of those reductions.

The actual grep count on the rebased working tree — `grep -roh --include="*.tsx" --include="*.ts" --include="*.jsx" --include="*.js" -E '\b[a-z][a-z0-9]*(-[a-z0-9]+)*-\[[^]]+\]' apps/web/components apps/web/app | wc -l` — is **5066**, verified against the actual files at `41c29b8`. The ratchet assertion will be `expect(5066).toBeLessThanOrEqual(5066)` → ✅.

Setting the baseline to `5392` (the bot's suggestion) would silently allow 326 regressions to creep back into the codebase.

---
_Generated by [Claude Code](https://claude.ai/code)_
